### ISSUE TITLE
Unable to upload jpg when Create Placeholder Silhouettes is enabled

### ISSUE THREAD

**Issue by tylerbrostrom** [nystudio107/craft3-imageoptimize#42]
### Description

Getting the following error message when attempting to upload a jpg (sized 2000 × 1239) to an asset volume with an attached `OptimizedImages` field:
```
Upload failed. The error message was: “Call to undefined function nystudio107\imageoptimize\lib\imagecreatefromjpeg()”
```
Looks like there's a `imagecreatefromjpeg` call in `./vendor/nystudio107/craft-imageoptimize/src/lib/Potracio.php`. Setting `createPlaceholderSilhouettes => false` suppresses the upload error.

### Additional Info

The attached `OptimizedImages` field has only one image variant. It's configured like:

| Field Setting | Value |
|-|-|
| **Width** | 1440 |
| **Enforce Aspect Ratio** | false |
| **Retina Sizes** | none |
| **Quality** | Very High |
| **Image Format** | Auto |

 Composer dependencies look okay:
`composer install` returns "Nothing to install or update"
`composer status` returns "No local changes"

ImageOptimize is using Craft native transforms with the following libs successfully installed:

| Optimization Libs |
|-|
| jpegoptim |
| optipng |
| svgo |
| gifsicle |
| cwepb |

| Software | Version |
|-|-|
| **ImageOptimize** | 1.4.13 |
| **Craft** | Craft Pro 3.0.0-RC9 |
| **PHP** | 7.0.22 |
| **Image driver** | Imagick 3.4.0RC6 (ImageMagick 6.8.9-9) |
| **Linux distribution** | Ubuntu 16.04.2 LTS |

**Comment by khalwat** [comment]
This sounds like it might be an issue with your local PHP install. That function is part of PHP:

http://php.net/manual/en/function.imagecreatefromjpeg.php

Maybe you don't have GD installed?

https://stackoverflow.com/questions/13338339/imagecreatefromjpeg-and-similar-functions-are-not-working-in-php

**Comment by tylerbrostrom** [comment]
@khalwat Oh boy… you're certainly correct.

Craft's [server requirements doc](https://github.com/craftcms/docs/blob/v3/en/requirements.md#required-php-extensions) states that either GD or ImageMagick need to be installed, ImageMagick being the preferred tool. Being the minimalist that I am, I had only installed ImageMagick.

Apologies for the false issue, and as always thanks for the help!

**Comment by khalwat** [comment]
Yeah it's a library that I'm using that apparently uses GD directly. I can see about eliminating that dependency.

**Comment by khalwat** [comment]
Addressed in 1.4.14; it will no longer attempt to generate a placeholder silhouette if you do not have `GD` installed, and it will give you a warning message on the Settings page to that effect:

https://github.com/nystudio107/craft-imageoptimize/releases/tag/1.4.14
